libs/indexer/map_object.hpp

```diff
@@ -14,25 +14,59 @@
 
 namespace osm
 {
-class EditableMapObject;
+
+// Wrapper over StringUtf8Multilang for additional constrains:
+// - Don't store empty names as-is, remove them instead
+// - Check supported languages
+// - Remove trailing whitespaces
+class FeatureNames
+{
+  StringUtf8Multilang m_str;
+
+public:
+  /// @name Actually, from/to FeatureTyoe.
+  /// @{
+  void FromBuffer(StringUtf8Multilang const & str) { m_str = str; }
+  StringUtf8Multilang const & ToBuffer() const { return m_str; }
+  /// @}
+
+  void Add(int8_t langCode, std::string_view name);
+  void Add(std::string_view lang, std::string_view name);
+
+  std::string_view Get(int8_t langCode) const;
+
+  template <class FnT>
+  void ForEach(FnT && fn) const
+  {
+    m_str.ForEach(fn);
+  }
+
+  /// @todo String may be different only because of the langs order inside a buffer.
```

Comment:
Does it make sense to force the order when a string is added to the buffer (e.g. with a separate method)?